rrs/tools/rrs_maintainer_history.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+
+# Standalone script which rebuilds the history of maintainership
+#
+# Copyright (C) 2015 Intel Corporation
+# Author: Anibal Limon <anibal.limon@linux.intel.com>
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os.path
+import optparse
+import logging
+
+sys.path.insert(0, os.path.realpath(os.path.join(os.path.dirname(__file__))))
+from common import common_setup, get_logger, DryRunRollbackException
+common_setup()
+from layerindex import utils, recipeparse
+
+utils.setup_django()
+from django.db import transaction
+import settings
+
+from layerindex.models import Recipe, LayerBranch, LayerItem
+from rrs.models import MaintenancePlan, Maintainer, RecipeSymbol, RecipeMaintainerHistory, RecipeMaintainer, RecipeMaintenanceLink
+from django.core.exceptions import ObjectDoesNotExist
+
+# FIXME we shouldn't be hardcoded to expect RECIPE_MAINTAINER to be set in this file,
+# as it may be in the recipe in future
+MAINTAINERS_INCLUDE_PATH = 'conf/distro/include/maintainers.inc'
+
+
+"""
+    Try to get recipe maintainer from line, if not found return None
+"""
+def get_recipe_maintainer(line, logger):
+    import re
+    regex = re.compile('^RECIPE_MAINTAINER_pn-(?P<pn>.*)\s=\s"(?P<name>.+) <(?P<email>.*)>"$')
+
+    match = regex.search(line)
+    if match:
+        return (match.group('pn'), match.group('name'), match.group('email'))
+    else:
+        logger.debug("line (%s) don\'t match" % (line))
+        return None
+
+"""
+    Get commit information from text.
+    Returns author_name, author_email, date and title.
+"""
+def get_commit_info(info, logger):
+    import re
+    from datetime import datetime
+    from email.utils import parsedate_tz, mktime_tz
+
+    author_regex = re.compile("^Author: (?P<name>.*) <(?P<email>.*)>$")
+    date_regex = re.compile("^Date:   (?P<date>.*)$")
+    title_regex = re.compile("^    (?P<title>.*)$")
+
+    lines = info.split('\n')
+
+    author_name = author_regex.search(lines[1]).group('name')
+    author_email = author_regex.search(lines[1]).group('email')
+    date_str = date_regex.search(lines[2]).group('date')
+    date = datetime.utcfromtimestamp(mktime_tz(parsedate_tz(date_str)))
+    title = title_regex.search(lines[4]).group('title')
+
+    return (author_name, author_email, date, title)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)